src/Twig/TwigExtension.php

```diff
@@ -55,4 +56,41 @@ public function dateNullableFilter($timestamp, $format, $nullDisplayValue = ''):
     {
         return $timestamp ? twig_date_format_filter($this->twig, $timestamp, $format) : $nullDisplayValue;
     }
+
+    /**
+     * Checks whether document on case is deletable.
+     */
+    public function isDocumentDeletable(Document $document, CaseEntity $case): bool
+    {
+        $digitalPosts = $this->digitalPostRepository->findByEntity($case);
+
+        // Check whether document has been sent out via digital post
+        foreach ($digitalPosts as $digitalPost) {
+            if ($document->getId() === $digitalPost->getDocument()->getId()) {
+                return false;
+            }
+
+            $attachments = $digitalPost->getAttachments();
+
+            foreach ($attachments as $attachment) {
+                if ($attachment->getDocument()->getId() === $document->getId()) {
+                    return false;
+                }
+            }
+        }
+
+        // Check whether document is attached to agenda case item
+        $agendaCaseItems = $case->getAgendaCaseItems();
+
+        foreach ($agendaCaseItems as $agendaCaseItem) {
+            $agendaCaseItemDocuments = $agendaCaseItem->getDocuments();
+            foreach ($agendaCaseItemDocuments as $agendaCaseItemDocument) {
+                if ($document->getId() === $agendaCaseItemDocument->getId()) {
+                    return false;
+                }
+            }
+        }
```

Comment:
It must/should be possible to do something like

```php
$items = $someRepository->findAgendaCaseItemsByDocument(…);
if (count($items) > 0) {
  return false;
}
```

Reply:
Yes - updated to use repository methods.